### ISSUE TITLE
Have some questions relating to multiple agents (single brain) training using camera

### ISSUE THREAD

**Issue by UsaidPro** [Unity-Technologies/ml-agents#396]
Hello! I've been poking around the unity-environment examples and it appears to me that the multiple arenas/areas used for the Area, Push, and 3DBall examples are useful because it lets the brain get the experiences of multiple different environments during a single step. Is this understanding correct?

Also, how would I use this for observations (cameras) as primary inputs to the Brain rather than states?Looking at all the examples that involve multiple agents (with one brain) none use observations and instead use states. My current understanding is that an unique agent ties its unique "states" and "actions" to a Brain but not observations (observations aren't "unique" to an agent) and therefore a Brain trained with multiple areas and multiple observations per area would require multiple observations even when there is a single area. 

And finally, under what circumstances should I use a state over an observation (which conveys much more)? Sorry about the incomplete title - I clicked "Submit" without checking it.

**Comment by vincentpierre** [comment]
Hi !
 * You are correct, having multiple agents in the scene allows ML-Agents to collect more data points per step which can make training a lot faster.
 * The visual observations are per agent. You can give each agent a different camera and it will use that input. You will need one camera per agent (or area). Note that having a lot of cameras will slow down your Unity Environment a lot.
 * States are easier to process for a neural network. And since you know everything about your game state via script, vector observation should be able to carry a lot more information than a visual observation.
 * You can still edit the title of a github issue by clicking the Edit button next to your issue title on the issue page.

I hope this helps !

**Comment by UsaidPro** [comment]
First of all, thank you for your incredibly fast reply @vincentpierre! I'll try to use multiple agents then. 

My project is basically a "critter" (agent) trying to get to a "food" (goal) in a grid world setting. I have tested using states and observations with results that confirm what you stated. Using a Brain that took the position of the agent and the goal as inputs performed much better (converged 3x faster) than an observation of the entire field. Using both only had a marginal improvement over just states.

Despite this, I think I need to use observations rather than states to convey the positions of the critter/food because I'm  planning to have multiple critters/food which would mean that the number of "food" positions would vary with critters "eating" food. I tried to pass in the state of every grid cell in the world as a 1xN array, but the PPO optimizer failed to improve. It's there a better way to represent the grid as states? Or should I use observations, seeing as there will be varying numbers of food and critters?

**Comment by vincentpierre** [comment]
I think it would help you to have a look at the banana environment in the branch development-0.3. It sound very similar to what you are tying to accomplish. I would recommend using raycasts instead of visual observation, it combines the simplicity of using float values and the local observations that the cameras provide.

**Comment by UsaidPro** [comment]
I managed to create 8 different areas for training, and I saw a very obvious increase in training! Training that normally took over 1000000 iterations took 10000 to get to the same reward.

I'm still using the camera though and want to move to raycasts. I have looked at the "banana" environment provided and noticed that it uses Vector Observations, which doesn't seem to be supported by the current (master) branch and was added to the development branch. Should I simply feed it in as a state for now? **What would be the difference between using observation vs using states?** Would supplying a entire grid as a state be the same as giving that same grid via a camera observation?
